### PR TITLE
karma-preset: Invalid path when using objects in files

### PR DESCRIPTION
**Pull request by mblarsen**
With a files config like this:

```
files: [
  'test/*.test.js',
  {pattern: 'test/*.m??', watched: false, included: false, served: true, nocache: false},
]
```

You would end up with a preprocessor array like this (see the 2nd key):

```
  preprocessors:
   { 'test/*.test.js': [ 'webpack', 'sourcemap' ],
     '[object Object]': [ 'webpack', 'sourcemap' ] },
```

This PR checks type and if object only includes if `included: false` (or absent)

### CODE REVIEW

File: packages/poi-preset-karma/index.js [hunk 1]
```diff
@@ -57,7 +57,11 @@ module.exports = (options = {}) => {
           }]
         },
         preprocessors: files.reduce((current, next) => {
-          current[next] = ['webpack', 'sourcemap']
+          if (typeof next === 'object' && Boolean(next.included)) {
```
Comment (egoist) [inline packages/poi-preset-karma/index.js]: what if `included` is absent here? it should be included as well since it defaults to `true`

Comment (mblarsen) [reply packages/poi-preset-karma/index.js]: If absent it will not be included. Let me look up the default value. Thought it to be true.

Comment (mblarsen) [reply packages/poi-preset-karma/index.js]: Yeah the default value is true, so it should be good.

```
included
Type. Boolean
Default. true
```

[docs](http://karma-runner.github.io/1.0/config/files.html)

Comment (egoist) [reply packages/poi-preset-karma/index.js]: if it's absent `Boolean(next.included)` will be false 😅

Comment (mblarsen) [reply packages/poi-preset-karma/index.js]: Doh, sorry -- will correct that (and have some coffee)

Comment (egoist) [reply packages/poi-preset-karma/index.js]: hah we can simply include it by checking `next.included !== false` I guess

Comment (mblarsen) [reply packages/poi-preset-karma/index.js]: Yes, that is correct

Comment (mblarsen) [reply packages/poi-preset-karma/index.js]: I don't think people would use other truthy or falsy values than true/false.

**Comment by egoist** [comment]
hmm btw why would you use an object in files?

**Pull request merged**